### ISSUE TITLE
[feature] [byob] Allow project checkout to be customized

### ISSUE THREAD

**Issue by aalmiray** [slsa-framework/slsa-github-generator#2043]
Both [goreleaser](https://github.com/goreleaser/goreleaser-action) and [jreleaser](https://github.com/jreleaser/release-action) actions require a full checkout (`fetch-depth: 0`) but the current `secure-project-checkout` sets `fetch-depth` to `1`. Allow `fetch-depth` to be specified as an optional parameter.

https://github.com/slsa-framework/slsa-github-generator/blob/6369788cef3f2103297351ee2e0605034a3f51f0/.github/actions/secure-project-checkout/action.yaml#L18-L25

Also, there may be times when the ref may not exactly match the original ref found in the `github` scope. This may happen when the PW creates a commit before invoking the TRW. Thus, the `ref` should also be passed as an optional parameter.

**Comment by ianlewis** [comment]
Thanks for the feedback!

This seems like two issues, one regarding `fetch-depth` and one regarding the `ref`. I think `fetch-depth` could just always be zero, though perhaps it would be better for it to be passed as an option for large repos?

WRT the `ref`, one of the reasons we have the `secure-project-checkout` is so that `repo` and `ref` are not under the PW's or even TRW's control. We may need to rethink this if release tools are creating/modifying commits. Can we assume the new commits that the PW makes are pushed to the repo before calling the TRW? Do you have an example PW you can link us to?

/cc @laurentsimon

**Comment by aalmiray** [comment]
Yes, the assumption is that commits made by the PW are pushed before invoking the TRW. Here's an example of a Maven based project that creates a commit before invoking JReleaser. Ideally this explicit invocation of the JReleaser action would be substituted with an invocation to JReleaser TRW.

https://github.com/kordamp/pomchecker/blob/ca5ae449d6b599c6461c801db04f00bfa0e94ff4/.github/workflows/release.yml#L44-L54

**Comment by laurentsimon** [comment]
Closing this issue as completed. Last AI is to add more test, tracked in https://github.com/slsa-framework/slsa-github-generator/issues/2079
